### PR TITLE
Add cuda_custom_kernel in DETA

### PR DESCRIPTION
**Pull request by SangbumChoi**
# What does this PR do?

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [X] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@amyeroberts Hi, can you check this PR? This will make training DETA with custom_cuda_kernel same in Deformable DETR!
I also confirmed that training with this kernel working properly in custom dataset

### CODE REVIEW

**Review by amyeroberts — COMMENTED** [review]
Thanks for adding this - looks great! 

Just some small structural comments

File: src/transformers/models/deta/load_custom.py
[the remark this answers is not in the document]
Comment (SangbumChoi) [reply src/transformers/models/deta/load_custom.py:1]: @amyeroberts this is copied from deformable detr should I move the modeling_deta.py?

[the remark this answers is not in the document]
Comment (amyeroberts) [reply src/transformers/models/deta/load_custom.py:1]: Ah, I see. So in other models, we keep this in the modeling file e.g. like [here for yoso](https://github.com/huggingface/transformers/blob/0507e69d34f8902422eb4977ec066dd6bef179a0/src/transformers/models/yoso/modeling_yoso.py#L53). I think this is better and more in-line with our one file per-model policy.

[the remark this answers is not in the document]
Comment (SangbumChoi) [reply src/transformers/models/deta/load_custom.py:1]: Understood!

[the remark this answers is not in the document]
Comment (SangbumChoi) [reply src/transformers/models/deta/load_custom.py:1]: @amyeroberts I changed like as yoso format! I can change deformable detr also if it is necessary.

[the remark this answers is not in the document]
Comment (amyeroberts) [reply src/transformers/models/deta/load_custom.py:1]: Yes please - that would be great!

**Review by amyeroberts — APPROVED** [review]
Amazing work - thanks for adding!

**Pull request merged**